src/components/ScValPrettyJson/index.tsx

```diff
@@ -79,269 +72,322 @@ export const ScValPrettyJson = ({
           };
 
       return (
-        <div className="ScValPrettyJson__value" data-type="address">
-          <SdsLink {...params} variant="secondary" isUnderline title={value}>
-            {shortenStellarAddress(value)}
-          </SdsLink>
-        </div>
+        <SdsLink {...params} variant="secondary" isUnderline title={value}>
+          {shortenStellarAddress(value)}
+        </SdsLink>
       );
     }
 
+    return shortenStellarAddress(value);
+  };
+
+  const renderPrimitive = ({
+    value,
+    type,
+    isValueOnly,
+    hasComma = false,
+  }: {
+    value: any;
+    type: string;
+    isValueOnly?: boolean;
+    hasComma?: boolean;
+  }) => {
+    switch (type) {
+      case "address":
+        return (
+          <Value key={`${type}-${uuidv4()}`} type={type} hasComma={hasComma}>
+            {renderAddress(value)}
+          </Value>
+        );
+      case "i8":
+      case "u8":
+      case "i16":
+      case "u16":
+      case "i32":
+      case "u32":
+      case "i64":
+      case "u64":
+        return isValueOnly ? (
+          stringify(value)
+        ) : (
+          <Value key={`${type}-${uuidv4()}`} type={type} hasComma={hasComma}>
+            {stringify(value)}
+          </Value>
+        );
+      case "i128":
+      case "u128":
+      case "i256":
+      case "u256": {
+        let val = stringify(value);
+
+        try {
+          const hiVal = stringify((value as any)?.hi);
+          const loVal = stringify((value as any)?.lo);
+
+          if (hiVal && loVal) {
+            val = stringify(new XdrLargeInt(type, [loVal, hiVal]).toBigInt());
+          }
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        } catch (e) {
+          // do nothing
+        }
+
+        return isValueOnly ? (
+          val
+        ) : (
+          <Value key={`${type}-${uuidv4()}`} type={type} hasComma={hasComma}>
+            {val}
+          </Value>
+        );
+      }
+      case "symbol":
+        if (typeof value === "object") {
+          return render({ item: value, parentType: type, hasComma });
+        }
+
+        if (isValueOnly) {
+          return value;
+        }
+
+        return (
+          <Value key={`${type}-${uuidv4()}`} type={type} hasComma={hasComma}>
+            {value}
+          </Value>
+        );
+      case "bool":
+      case "string":
```

Comment:
can add a case for `dataurl (or bytes)` as well ([doc](https://developers.stellar.org/docs/learn/encyclopedia/contract-development/types/built-in-types#bytes-strings-bytes-bytesn-string))?

this is what stellar.expert does:
![Screenshot 2025-05-20 at 11 44 41 AM](https://github.com/user-attachments/assets/d4a7ed9d-3c84-4b7a-9288-3474dadd75de)

Reply:
Looks like they are strings, so they should be handled by the default case. I don't think we need to list all the types, unless we need to handle them differently.

Reply:
Sorry, accidentally clicked resolved when I wanted to upload an image. 

I think there's benefit in distinguish between each string types. Today we show differet types with different colors.
<img width="1009" alt="Screenshot 2025-05-22 at 8 39 04 PM" src="https://github.com/user-attachments/assets/ae49e17a-620b-488f-be7f-b9aef5a9d7ae" />